source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -741,101 +741,55 @@ namespace
         return destNode;
     }
 
-    void createMaterialNode(const PvtPrim* prim, NodePtr mxNode, DocumentPtr doc)
+    void writeMaterialElement(NodePtr mxNode, DocumentPtr doc, const RtWriteOptions* writeOptions)
     {
-        // Check to see if the surfaceshader node is already connected to a material node
-        bool isConnectedToMaterialNode = false;
-        const PvtOutput* output = prim->getOutput(PvtAttribute::DEFAULT_OUTPUT_NAME);
-        if (output && output->getType() == RtType::SURFACESHADER && output->isConnected())
-        {
-            for (const RtObject& dest : output->getConnections())
-            {
-                RtInput destInput = dest.asA<RtInput>();
-                RtPrim destNode = destInput.getParent();
-                RtOutput destOutput = destNode.getOutput(PvtAttribute::DEFAULT_OUTPUT_NAME);
-                if (destOutput.getType() == RtType::MATERIAL)
-                {
-                    isConnectedToMaterialNode = true;
-                    break;
-                }
-            }
-        }
+        string uniqueName = doc->createValidChildName(mxNode->getName() + "_Material");
+        string materialName = mxNode->getName();
+        mxNode->setName(uniqueName);
 
-        if (!isConnectedToMaterialNode)
+        InputPtr materialNodeSurfaceShaderInput = mxNode->getInput(RtType::SURFACESHADER);
+        NodePtr surfaceShader = materialNodeSurfaceShaderInput->getConnectedNode();
+        if (surfaceShader)
         {
-            NodePtr materialNode = doc->addNode(SURFACE_MATERIAL_NODE_STRING, mxNode->getName() + "_SurfaceMaterial", MATERIAL_TYPE_STRING);
-            materialNode->setConnectedNode(SURFACE_SHADER_TYPE_STRING, mxNode);
-        }
-    }
-
-    void writeMaterialElementsHelper(const PvtPrim* prim, NodePtr mxNode, const string& materialBaseName, DocumentPtr doc, const RtWriteOptions* writeOptions)
-    {
-        MaterialPtr material = doc->addMaterial(materialBaseName + "_Material");
-        ShaderRefPtr shaderRef =
-            material->addShaderRef("sref", mxNode->getCategory());
+            MaterialPtr material = doc->addMaterial(materialName);
+            ShaderRefPtr shaderRef =
+                material->addShaderRef("sref", surfaceShader->getCategory());
```

Comment:
That's what I thought originally as well, but Pavlo pointed out that isn't correct, this is.

Reply:
Yep, please refer to #752. A material element should reference e.g. `standard_surface` instead of the specific shader's name. #752 fixed MtoA deserialization for me, so it has been verified.